pkg/util/docker/containers.go

```diff
@@ -250,6 +280,24 @@ func isExposed(port types.Port) bool {
 	return port.PublicPort > 0 && port.IP != ""
 }
 
+// getECSMetadataURL inspects a given container ID and returns its ECS container metadata URI
+// if found in its environment. It returns an empty string and an error on failure.
+func (d *DockerUtil) getECSMetadataURL(cID string) (string, error) {
+	i, err := d.Inspect(cID, false)
+	if err != nil {
+		return "", err
+	}
+	for _, e := range i.Config.Env {
+		split := strings.Split(e, "=")
```

Comment:
Nit: you could use strings.HasPrefix to avoid an allocation for every environment variable.

Reply:
that's a good point!